src/Config.cpp

```diff
@@ -223,13 +222,296 @@ bool readEndCommandFromEEPROM(uint16_t *addreeprom)
         temp = MFeeprom.read_byte((*addreeprom)++);
         if (*addreeprom > length) // abort if EEPROM size will be exceeded
             return false;
-    } while (temp != ':');        // reads until limiter ':'
+    } while (temp != ':'); // reads until limiter ':'
+    return true;
+}
+
+bool getArraysizes()
+{
+    if (configLength == 0) // do nothing if no config is available
+        return true;
+    uint16_t addreeprom          = MEM_OFFSET_CONFIG;               // define first memory location where config is saved in EEPROM
+    char     command             = readUintFromEEPROM(&addreeprom); // read the first value from EEPROM, it's a device definition
+    bool     copy_success        = true;                            // will be set to false if copying input names to nameBuffer exceeds array dimensions
+    uint8_t  numberButtons       = 0;
+    uint8_t  numberEncoders      = 0;
+    uint8_t  numberOutputs       = 0;
+    uint8_t  numberLedSegments   = 0;
+    uint8_t  numberStepper       = 0;
+    uint8_t  numberServos        = 0;
+    uint8_t  numberLCD           = 0;
+    uint8_t  numberAnalogIn      = 0;
+    uint8_t  numberOutputShifter = 0;
+    uint8_t  numberInputShifter  = 0;
+    uint8_t  numberDigInMux      = 0;
+
+    if (command == 0) // just to be sure, configLength should also be 0
+        return true;
+
+    // go through the EEPROM and calculate the number of devices for each type
+    do // go through the EEPROM until it is NULL terminated
+    {
+        switch (command) {
```

Comment:
The "switch" could probably be avoided by placing the counter in an array indexed with the `command` value (as long as the `kTypeXxxxx` values are reasonably not sparse.
The whole case could disappear in something like `numberDevice[command]++;`

Reply:
The `kTypeXxxxx` values are not perfect for an array due to the deprecated devices, but it makes still sense.

Reply:
> The `kTypeXxxxx` values are not perfect for an array due to the deprecated devices, but it makes still sense.

Exactly; however, each "useless" element would only cost 1 byte of RAM, while the total flash gain could be more interesting (still to be verified).